### PR TITLE
feat(python): Add gemini text embedding function

### PR DESCRIPTION
**Pull request by AyushExel**
Named it Gemini-text for now. Not sure how complicated it will be to support both text and multimodal embeddings under the same class "gemini"..But its not something to worry about for now I guess.

### CODE REVIEW

File: python/lancedb/embeddings/gemini_text.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        Sets the task type for ingestion.
+
+    Examples
+    --------
+    import lancedb
+    import pandas as pd
+    from lancedb.pydantic import LanceModel, Vector
+    from lancedb.embeddings import get_registry
+
+    def test_instructor_embedding(tmp_path):
```
Comment (changhiskhan) [inline python/lancedb/embeddings/gemini_text.py:61]: again, copy-pasted unit test doesn't make sense here

File: python/lancedb/embeddings/gemini_text.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        ):  # Provide a title to use existing API design
+            title = "Embedding of a document"
+            kwargs["title"] = title
+
+        return [
+            genai.embed_content(model=self.name, content=text, **kwargs)["embedding"]
+            for text in texts
+        ]
+
+    def get_client(self):
```
Comment (changhiskhan) [inline python/lancedb/embeddings/gemini_text.py:121]: i think you can make this a cached property

File: docs/src/embeddings/default_embedding_functions.md [hunk 3]
```diff
@@ -118,6 +118,42 @@ texts = [{"text": "Capitalism has been dominant in the Western world since the e
...
+
+Usage Example:
+
+```python
+import lancedb
+import pandas as pd
+from lancedb.pydantic import LanceModel, Vector
+from lancedb.embeddings import get_registry
+
+def test_instructor_embedding(tmp_path):
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply docs/src/embeddings/default_embedding_functions.md]: Moved everything outside the function

**Comment by AyushExel** [comment]
@changhiskhan cached_property doesn't seem to work with pydantic 1.x, using lru_cache instead

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**